### PR TITLE
ts-717 separate integration tests from unit tests

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -22,12 +22,14 @@ import sbt._
...
-      "org.scalamock" %% "scalamock-scalatest-support" % "3.5.0" % "test",
-      "org.rogach" %% "scallop" % "3.1.1" % "test",
-      "org.apache.commons" % "commons-math3" % "3.6.1" % "test",
-      "org.mockito" % "mockito-core" % "2.13.0" % "test"
+      "org.scalatest" %% "scalatest" % "3.0.4" % "it,test",
+      "org.mockito" % "mockito-core" % "2.13.0" % "it,test",
+      "org.scalamock" %% "scalamock-scalatest-support" % "3.5.0" % "it,test",
+      "org.rogach" %% "scallop" % "3.1.1" % "it,test",
+      "org.apache.commons" % "commons-math3" % "3.6.1" % "it,test",
+      "org.mockito" % "mockito-core" % "2.13.0" % "it,test"
```
Comment (placccebo) [inline project/Dependencies.scala]: redundant dependency

Comment (tomskikh) [reply project/Dependencies.scala]: Removed

File: build.sbt [hunk 2]
```diff
@@ -27,10 +27,12 @@ lazy val root = (project in file("."))
...
-  .settings(Common.projectSettings ++ Dependencies.TStreams: _*)
+  .configs(IntegrationTest)
+  .settings(Common.projectSettings)
   .dependsOn(tStreamsTransactionServer)
 
 lazy val tStreamsTransactionServer = Project(
   id = "tstreams-transaction-server",
   base = file("./tstreams-transaction-server"))
+  .configs(IntegrationTest)
   .settings(TStreamsTransactionServer.projectSettings: _*)
```
Comment (bwsw) [inline build.sbt]: add new line here.

File: t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/ProducerCancelPartitionTest.scala [hunk 3]
```diff
@@ -56,15 +56,16 @@ class ProducerCancelPartitionTest  extends FlatSpec with Matchers with BeforeAnd
...
 
     producer.cancel(0) // cancel empty one
 
-    (0 until TOTAL_CANCEL).foreach(t => producer.newTransaction())
+    (0 until TOTAL_CANCEL).foreach(_ => producer.newTransaction())
     producer.cancel(0)
 
-    (0 until TOTAL_CHECKPOINT).foreach(t => producer.newTransaction().send("test"))
+    (0 until TOTAL_CHECKPOINT).foreach(_ => producer.newTransaction().send("test"))
+    (0 until TOTAL_CHECKPOINT).foreach(_ => producer.newTransaction().send("test"))
```
Comment (bwsw) [inline t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/ProducerCancelPartitionTest.scala]: Why the expression goes two times? Test is changed?

File: t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/TwoProducersToSubscriberStartsAfterWriteTests.scala [hunk 4]
```diff
@@ -113,7 +114,7 @@ class TwoProducersToSubscriberStartsAfterWriteTests extends FlatSpec with Matche
 
 
   override def afterAll(): Unit = {
-    TestStorageServer.dispose(srv)
+    TestStorageServer.dispose(server)
     onAfterAll()
   }
 }
```
Comment (bwsw) [inline t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/TwoProducersToSubscriberStartsAfterWriteTests.scala]: add new line

File: t-streams/src/it/scala/com/bwsw/tstreams/env/TStreamsFactoryTest.scala [hunk 5]
```diff
@@ -51,7 +51,7 @@ class TStreamsFactoryTest extends FlatSpec with Matchers with BeforeAndAfterAll
       n1.setProperty(ConfigurationOptions.Stream.name, "cloned-stream")
       false
     } catch {
-      case e: IllegalStateException =>
+      case _: IllegalStateException =>
         true
     }
     res shouldBe true
```
Comment (bwsw) [inline t-streams/src/it/scala/com/bwsw/tstreams/env/TStreamsFactoryTest.scala]: May be use intercept here?

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/CommonCheckpointGroupServerTest.scala [hunk 6]
```diff
@@ -194,7 +195,7 @@ class CommonCheckpointGroupServerTest
   }
 
   it should "[putProducerStateWithData] put a producer transaction (Opened) with data, and server should persist data." in {
-    val bundle = util.multiNode.Util.getCommonCheckpointGroupServerBundle(
+    val bundle = multiNode.Util.getCommonCheckpointGroupServerBundle(
```
Comment (bwsw) [inline tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/CommonCheckpointGroupServerTest.scala]: In the code there are Util, Utils, etc. Let's unify to Utils everywhere.

File: t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/ProducerCancelPartitionTest.scala (hunk 3 above)
Comment (tomskikh) [reply t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/ProducerCancelPartitionTest.scala]: It's mistake. Fixed.

File: t-streams/src/it/scala/com/bwsw/tstreams/env/TStreamsFactoryTest.scala (hunk 5 above)
Comment (tomskikh) [reply t-streams/src/it/scala/com/bwsw/tstreams/env/TStreamsFactoryTest.scala]: Yes, fixed.

File: tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/CommonCheckpointGroupServerTest.scala (hunk 6 above)
Comment (tomskikh) [reply tstreams-transaction-server/src/it/scala/com/bwsw/tstreamstransactionserver/netty/server/multiNode/CommonCheckpointGroupServerTest.scala]: Renamed

File: build.sbt (hunk 2 above)
Comment (tomskikh) [reply build.sbt]: Added

File: t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/TwoProducersToSubscriberStartsAfterWriteTests.scala (hunk 4 above)
Comment (tomskikh) [reply t-streams/src/it/scala/com/bwsw/tstreams/agents/integration/TwoProducersToSubscriberStartsAfterWriteTests.scala]: Added

**Comment by tomskikh** [comment]
[run tests]

**Comment by tomskikh** [comment]
[run tests]

**Pull request merged**